catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -1,513 +1,535 @@
+    raise
+
+
+class HamamatsuCamera(Service):
+    """
+    Service for Hamamatsu cameras.
+
+    This service is a wrapper around the DCAM-SDK4.
+    It provides a simple interface to control the camera and acquire images.
+
+    Attributes
+    ----------
+    cam : Camera
+        The camera to control.
+    pixel_formats : dict
+        A dictionary to store the pixel format and the corresponding numpy dtype and dcam pixel format.
+    current_pixel_format : str
+        The current pixel format to use.
+    temperature_thread : threading.Thread
+        A thread to monitor the temperature of the camera.
+    temperature : DataStream
+        A data stream to submit the temperature of the camera.
+    images : DataStream
+        A data stream to submit the images from the camera.
+    is_acquiring : DataStream
+        A data stream to submit whether the camera is currently acquiring images.
+    should_be_acquiring : threading.Event
+        An event to signal whether the camera should be acquiring images.
+    NUM_FRAMES : int
+        The number of frames to allocate for the data streams.
+    """
+    NUM_FRAMES = 20
+
+    def __init__(self):
+        """
+        Create a new HamamatsuCamera service.
+        """
+        super().__init__('hamamatsu_camera')
+
+        self.cam = None
+
+        # Dictionary to store the pixel format and the corresponding numpy dtype and dcam pixel format
+        self.pixel_formats = {
+            "Mono8": dcam.DCAM_PIXELTYPE.MONO8,
+            "Mono16": dcam.DCAM_PIXELTYPE.MONO16,
+        }
+        self.current_pixel_format = None
+        self.temperature_thread = None
+        self.temperature = None
+        self.images = None
+        self.is_acquiring = None
+
+        self.should_be_acquiring = threading.Event()
+        self.should_be_acquiring.set()
+
+    def open(self):
+        """
+        Open the service.
+
+        This function is called when the service is opened.
+        It initializes the camera and creates the data streams and properties.
+
+        Raises
+        ------
+        ValueError
+            If the pixel format is invalid.
+        RuntimeError
+            For a Dcamapi or Dcam error when initializing the library or when opening the camera.
+        """
+        if dcam.Dcamapi.init() is False:
+            raise RuntimeError(f'Dcamapi.init() fails with error {dcam.Dcamapi.lasterr()}')
+
+        camera_id = self.config.get('camera_id', 0)
+        self.cam = dcam.Dcam(camera_id)
+        self.log.info('Using camera with ID %s', camera_id)
+        if self.cam.dev_open() is False:
+            raise RuntimeError(f'Dcam.dev_open() fails with error {self.cam.lasterr()}')
+
+        # Set water cooling mode and fan mode
+        self.cooling_mode = self.config.get('cooling_mode', 'on')
+        self.fan_on = self.config.get('fan_on', True)
+
+        if self.fan_on:
+            fan = 1
+        else:
+            fan = 0
+

```

Comment:
If you want to be fancy, you can simplify this to: `fan = 1 if self.fan_on else 0`